module/data/item/templates/activated-effect.mjs

```diff
@@ -86,7 +87,8 @@ export default class ActivatedEffectTemplate extends foundry.abstract.DataModel
         per: new foundry.data.fields.StringField({
           required: true, nullable: true, blank: false, initial: null, label: "DND5E.LimitedUsesPer"
         }),
-        recovery: new FormulaField({required: true, label: "DND5E.RecoveryFormula"})
+        recovery: new FormulaField({required: true, label: "DND5E.RecoveryFormula"}),
+        prompt: new foundry.data.fields.BooleanField({initial: true, label: "DND5E.LimitedUsesPrompt"})
```

Comment:
It would be ideal if we could keep this PR focused purely on the refactor, and separate out this feature into its own PR once this one is merged, so that it can be discussed separately.

Reply:
I felt that we had agreed on it being useful but I'm happy to do that, lmk

Reply:
Yeah, I think there's an issue for it already that we can add to milestone if it's not already in it. I'd just prefer they be split into separate PRs.